### PR TITLE
feat: Payment card subscription size screen UI

### PR DESCRIPTION
**Pull request by MrMuzyk**
### Details

Implemented UI for subcription size screens

### Fixed Issues

$ https://github.com/Expensify/App/issues/38632
PROPOSAL:

### Tests

- go to `/settings/subscription/subscription-size`
- [x] Verify that no errors appear in the JS console

### QA Steps

- go to `/settings/subscription/subscription-size`
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-05-28 at 10 49 58
Screenshot 2024-05-28 at 10 50 41
Screenshot 2024-05-28 at 10 50 58
Screenshot 2024-05-28 at 10 51 13

</details>

### CODE REVIEW

File: src/ROUTES.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ const ROUTES = {
     SETTINGS_PRONOUNS: 'settings/profile/pronouns',
     SETTINGS_PREFERENCES: 'settings/preferences',
     SETTINGS_SUBSCRIPTION: 'settings/subscription',
+    SETTINGS_SUBSCRIPTIONS_SIZE: 'settings/subscription-size',
```
Comment (fabioh8010) [inline src/ROUTES.ts:92]: The route is different from what is described in the design doc, `/settings/subscription/subscription-size`.

File: src/languages/es.ts [hunk 2]
```diff
@@ -3610,4 +3610,21 @@ export default {
...
+        note: 'Nota: Un miembro activo es cualquiera que haya creado, editado, enviado, aprobado, reembolsado, o exportado datos de gastos vinculados al espacio de trabajo de tu empresa.',
+        confirmDetails: 'Confirma los datos de tu nueva suscripción anual',
+        subscriptionSize: 'Tamaño de suscripción',
+        activeMembers: ({size}) => `${size} miembros activos/mes`,
+        subscriptionRenews: 'Renovación de la suscripción',
+        youCantDowngrade: 'YNo puedes bajar de categoría durante tu suscripción anual',
+        youAlreadyCommitted: ({size, date}) =>
+            `Ya se ha comprometido a un tamaño de suscripción anual de ${size} miembros activos al mes hasta el ${date}. Puede cambiar a una suscripción de pago por uso en ${date} desactivando la auto-renovación.`,
+        error: {
+            size: 'Please enter a valid subscription size.',
```
Comment (fabioh8010) [inline src/languages/es.ts:3627]: Shouldn't this be in Spanish?

File: src/languages/es.ts [hunk 3]
```diff
@@ -3610,4 +3610,21 @@ export default {
     systemMessage: {
         mergedWithCashTransaction: 'encontró un recibo para esta transacción.',
     },
+    subscriptionSize: {
```
Comment (rezkiy37) [inline src/languages/es.ts:3613]: Please confirm translations in the `open-source` channel.

File: src/pages/settings/Subscription/SubscriptionSize/SubscriptionSizePage.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                onNext={nextScreen}
+                onMove={moveTo}
+            />
+        </ScreenWrapper>
+    );
+}
+
+SubscriptionSizePage.displayName = 'SubscriptionSizePage';
+
+export default withOnyx<SubscriptionSizePageProps, SubscriptionSizePageOnyxProps>({
```
Comment (fabioh8010) [inline src/pages/settings/Subscription/SubscriptionSize/SubscriptionSizePage.tsx:75]: Let's use `useOnyx` instead.

File: src/ROUTES.ts (hunk 1 above)
Comment (pasyukevich) [inline src/ROUTES.ts:92]: we have different address for this - settings/subscription/subscription-size

File: src/pages/settings/Subscription/SubscriptionSize/SubscriptionSizePage.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    subscriptionSizeForm: OnyxEntry<SubscriptionSizeForm>;
+};
+
+type SubscriptionSizePageProps = SubscriptionSizePageOnyxProps;
+
+const bodyContent: Array<React.ComponentType<SubStepProps>> = [Size, Confirmation];
+
+function SubscriptionSizePage({subscriptionSizeForm}: SubscriptionSizePageProps) {
+    const {translate} = useLocalize();
+    // TODO update logic to
```
Comment (fabioh8010) [inline src/pages/settings/Subscription/SubscriptionSize/SubscriptionSizePage.tsx:26]: Could we clarify more this TODO? e.g. when to remove it?

File: src/pages/settings/Subscription/SubscriptionSize/substeps/Confirmation.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                    text={translate(CAN_DOWNGRADE ? 'common.save' : 'common.close')}
+                />
+            </FixedFooter>
+        </View>
+    );
+}
+
+Confirmation.displayName = 'ConfirmationStep';
+
+export default withOnyx<ConfirmationProps, ConfirmationOnyxProps>({
```
Comment (fabioh8010) [inline src/pages/settings/Subscription/SubscriptionSize/substeps/Confirmation.tsx:76]: Let's use `useOnyx` instead.

File: src/pages/settings/Subscription/SubscriptionSize/substeps/Size.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import INPUT_IDS from '@src/types/form/SubscriptionSizeForm';
+
+type SizeProps = SubStepProps;
+
+function Size({onNext}: SizeProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+
+    const defaultValues = {
+        // TODO get default value from ONYX
```
Comment (fabioh8010) [inline src/pages/settings/Subscription/SubscriptionSize/substeps/Size.tsx:22]: Could we clarify more this TODO?

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+const validate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.SUBSCRIPTION_SIZE_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.SUBSCRIPTION_SIZE_FORM> => {
+    const errors = ValidationUtils.getFieldRequiredErrors(values, [INPUT_IDS.SUBSCRIPTION_SIZE]);
+    if (values[INPUT_IDS.SUBSCRIPTION_SIZE] && !ValidationUtils.isNumeric(values[INPUT_IDS.SUBSCRIPTION_SIZE])) {
+        errors.subscriptionSize = 'subscriptionSize.error.size';
+    }
+
+    return errors;
+};
+
+const getNewSubscriptionRenewalDate = (): string => format(startOfMonth(addMonths(new Date(), 11)), 'MMM d, yyyy');
```
Comment (fabioh8010) [inline src/pages/settings/Subscription/SubscriptionSize/utils.ts:16]:
```suggestion
const getNewSubscriptionRenewalDate = (): string => format(startOfMonth(addMonths(new Date(), 11)), CONST.DATE.MONTH_DAY_YEAR_ABBR_FORMAT);
```

File: src/ROUTES.ts (hunk 1 above)
Comment (MrMuzyk) [reply src/ROUTES.ts:92]: yeah, mb, fixing it!

File: src/languages/es.ts (hunk 2 above)
Comment (MrMuzyk) [reply src/languages/es.ts:3627]: Yes, I have to ask for the translation. It wasn't in the docs as validation wasn't covered at all

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts (hunk 8 above)
Comment (rezkiy37) [inline src/pages/settings/Subscription/SubscriptionSize/utils.ts:16]:
```suggestion
const getNewSubscriptionRenewalDate = (): string => format(startOfMonth(addMonths(new Date(), 11)), CONST.DATE.MONTH_DAY_YEAR_ABBR_FORMAT);
```

Comment (pasyukevich) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts:16]: I think we will need this method on other screens too 

wdyt?

**Review by rezkiy37 — CHANGES_REQUESTED** [review]
Left minor comments, besides LGTM.

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts (hunk 8 above)
Comment (MrMuzyk) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts:16]: Dont think so. In majority of the cases this date will come from ONYX, we will just have to format this. This helper prepares new date when editing subscription size

Comment (fabioh8010) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts:16]: Yeah, if it's going to be used in other places it's a good candidate to be on `SubscriptionUtils` for example

File: src/languages/es.ts (hunk 3 above)
Comment (MrMuzyk) [reply src/languages/es.ts:3613]: almost all of them are already confirmed

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts (hunk 8 above)
Comment (MrMuzyk) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts:16]: if such need arises then we can move it, is it ok to leave this here for now?

Comment (fabioh8010) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts:16]: > if such need arises then we can move it, is it ok to leave this here for now?

Up to you @pasyukevich 😉

Comment (pasyukevich) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts]: we can leave it here and move it up when it is needed

**Review by rezkiy37 — APPROVED** [review]

**Review by fabioh8010 — APPROVED** [review]

**Review by pasyukevich — APPROVED** [review]

File: src/ROUTES.ts [hunk 9]
```diff
@@ -89,6 +89,7 @@ const ROUTES = {
     SETTINGS_PRONOUNS: 'settings/profile/pronouns',
     SETTINGS_PREFERENCES: 'settings/preferences',
     SETTINGS_SUBSCRIPTION: 'settings/subscription',
+    SETTINGS_SUBSCRIPTIONS_SIZE: 'settings/subscription/subscription-size',
```
Comment (fabioh8010) [inline src/ROUTES.ts:92]:
```suggestion
    SETTINGS_SUBSCRIPTION_SIZE: 'settings/subscription/subscription-size',
```

@MrMuzyk Minor typo I noticed while working on my design doc

File: src/languages/es.ts [hunk 10]
```diff
@@ -3610,4 +3610,23 @@ export default {
...
+            note: 'Nota: Un miembro activo es cualquiera que haya creado, editado, enviado, aprobado, reembolsado, o exportado datos de gastos vinculados al espacio de trabajo de tu empresa.',
+            confirmDetails: 'Confirma los datos de tu nueva suscripción anual',
+            subscriptionSize: 'Tamaño de suscripción',
+            activeMembers: ({size}) => `${size} miembros activos/mes`,
+            subscriptionRenews: 'Renovación de la suscripción',
+            youCantDowngrade: 'No puedes bajar de categoría durante tu suscripción anual',
+            youAlreadyCommitted: ({size, date}) =>
+                `Ya se ha comprometido a un tamaño de suscripción anual de ${size} miembros activos al mes hasta el ${date}. Puede cambiar a una suscripción de pago por uso en ${date} desactivando la auto-renovación.`,
+            error: {
+                size: ' Por favor ingrese un tamaño de suscripción valido.',
```
Comment (ikevin127) [inline src/languages/es.ts:3628]: @MrMuzyk Is this space at the beginning of the `error.size` intentional ?

**Comment by ikevin127** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/56457735/69454066-c42d-46b1-8db5-0f74fe69421b



</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/56457735/87ac30a4-2aff-48d5-be32-b3057023b6a4



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/56457735/cc4f989e-798f-4fb3-aaef-12ce03bbc6fe



</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/56457735/49333ac3-f5fd-4eb5-ab41-c108fae533c4



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/56457735/916e09af-d4e2-4087-bd0f-6c0c342021d1



</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/56457735/2e6fca58-3d27-437f-a98b-46f3cb6fb705




</details>

**Review by ikevin127 — APPROVED** [review]
UI looks good to me! Currently only accessible via route URL.

Approved with the mention of https://github.com/Expensify/App/pull/42683#discussion_r1619300041 where I noticed spanish translation empty space at the beginning of a string.

**Comment by trjExpensify** [comment]
Nice one, Kevin and Melvin! @amyevans will you review this one?

**Comment by amyevans** [comment]
Ah yep! Will get to it tonight although I have to step away for a few hours at the moment

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import {addMonths, format, startOfMonth} from 'date-fns';
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import CONST from '@src/CONST';
+import type ONYXKEYS from '@src/ONYXKEYS';
+import INPUT_IDS from '@src/types/form/SubscriptionSizeForm';
+
+const validate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.SUBSCRIPTION_SIZE_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.SUBSCRIPTION_SIZE_FORM> => {
+    const errors = ValidationUtils.getFieldRequiredErrors(values, [INPUT_IDS.SUBSCRIPTION_SIZE]);
+    if (values[INPUT_IDS.SUBSCRIPTION_SIZE] && !ValidationUtils.isNumeric(values[INPUT_IDS.SUBSCRIPTION_SIZE])) {
```
Comment (amyevans) [inline src/pages/settings/Subscription/SubscriptionSize/utils.ts:10]: We should also disallow `0` - not sure if you'd like to handle here or if it'll be covered in a future PR

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+        errors.subscriptionSize = 'subscription.subscriptionSize.error.size';
+    }
+
+    return errors;
+};
+
+const getNewSubscriptionRenewalDate = (): string => format(startOfMonth(addMonths(new Date(), 11)), CONST.DATE.MONTH_DAY_YEAR_ABBR_FORMAT);
```
Comment (amyevans) [inline src/pages/settings/Subscription/SubscriptionSize/utils.ts:17]:
```suggestion
const getNewSubscriptionRenewalDate = (): string => format(startOfMonth(addMonths(new Date(), 12)), CONST.DATE.MONTH_DAY_YEAR_ABBR_FORMAT);
```
12 months matches the backend logic

**Review by amyevans — CHANGES_REQUESTED** [review]
Two quick comments otherwise LGTM!

Also please update the base branch to `main`

File: src/languages/es.ts (hunk 10 above)
Comment (MrMuzyk) [reply src/languages/es.ts:3667]: nope, fixing it

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts (hunk 12 above)
Comment (MrMuzyk) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts:17]: Changing it. I've found `11` in doc. Must've been a mistake then
Screenshot 2024-06-03 at 09 59 56

**Comment by shawnborton** [comment]
For the second paragraph here:
CleanShot 2024-06-03 at 09 57 10@2x

We might as well make it in the same font size as the first paragraph. Otherwise it feels weird that it is smaller than the other but still uses the same color. cc @Expensify/design for thoughts.


For the small text under the input here
CleanShot 2024-06-03 at 09 57 51@2x

This feels like hint text, and we should make it follow the same pattern we have elsewhere for hint text under an input: move it closer to the input, and make it use our textSupporting color

I left similar comments in the design doc since I think we missed that part of the design review, but feel free to continue the conversation here

File: src/pages/settings/Subscription/SubscriptionSize/utils.ts (hunk 11 above)
Comment (MrMuzyk) [reply src/pages/settings/Subscription/SubscriptionSize/utils.ts:10]: I will change it now

**Comment by MrMuzyk** [comment]
@shawnborton Would you like me to update styles of these texts now or should I wait for thoughts from design team you've CCed?

**Comment by shawnborton** [comment]
I have a feeling they will likely agree with me :) so I think it's probably okay to start updating them.

**Comment by MrMuzyk** [comment]
Screenshot 2024-06-03 at 11 19 22
Screenshot 2024-06-03 at 11 19 35
Screenshot 2024-06-03 at 11 21 01

Does it look ok now?

**Comment by shawnborton** [comment]
That looks great to me!

**Review by amyevans — APPROVED** [review]
LGTM!

@ikevin127 can you give the latest changes a quick review once you're online?

**Comment by dannymcclain** [comment]
Thanks Shawn! I took care of the mocks in the doc, but glad it's gotten sorted here! And as you expected, I 100% agree with all your suggested updates 🤝

**Review by ikevin127 — APPROVED** [review]
Matches the latest design requirements, let's go! 🚀

**Review by amyevans — APPROVED** [review]

File: src/pages/settings/Subscription/SubscriptionSize/substeps/Size.tsx [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+                <Text style={[styles.textNormalThemeText, styles.mb5]}>{translate('subscription.subscriptionSize.yourSize')}</Text>
+                <InputWrapper
+                    InputComponent={TextInput}
+                    inputID={INPUT_IDS.SUBSCRIPTION_SIZE}
+                    label={translate('subscription.subscriptionSize.subscriptionSize')}
+                    aria-label={translate('subscription.subscriptionSize.subscriptionSize')}
+                    role={CONST.ROLE.PRESENTATION}
+                    defaultValue={defaultValues[INPUT_IDS.SUBSCRIPTION_SIZE]}
+                    shouldSaveDraft
+                />
```
Comment (ahmedGaber93) [inline src/pages/settings/Subscription/SubscriptionSize/substeps/Size.tsx:44]: coming from https://github.com/Expensify/App/issues/69011, for better UX, we add `inputMode={CONST.INPUT_MODE.NUMERIC}` to this input to open the correct keyboard type.

**Pull request merged**